python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -64,10 +64,20 @@ def __getattribute__(self, key):
             raise
 
     def __iter__(self):
-        group_names, offsets, grouped_values = self._grouped()
+        group_names, offsets, grouped_keys, grouped_values = self._grouped()
         for i, name in enumerate(group_names):
             yield name, grouped_values[offsets[i] : offsets[i + 1]]
 
+    @cached_property
+    def groups(self):
+        """
+        Returns a dictionary mapping group keys to row labels.
+        """
+        group_names, offsets, _, _ = self._grouped()
+        return dict(
+            zip(group_names.to_pandas(), self.obj.index._split(offsets[1:-1]))
```

Comment:
Is this what we should return?

Reply:
cc: @kkraus14

Reply:
Yes this is what we should return, but I think we want to add a heuristic to this and other iteration based methods that over say 100 or 1000 groups there's going to be significant performance degradation.

Reply:
👍